src/ui/ui_menu_nbgl.c

```diff
@@ -0,0 +1,110 @@
+#include "glyphs.h"
+#include "ux.h"
+#include "nbgl_use_case.h"
+#include "settings.h"
+
+static void displaySettingsMenu(void);
+static void settingsControlsCallback(int token, uint8_t index);
+static bool settingsNavCallback(uint8_t page, nbgl_pageContent_t* content);
+
+enum { SWITCH_HASH_SET_TOKEN = FIRST_USER_TOKEN, SWITCH_SEQUENCE_SET_TOKEN };
+
+#define NB_INFO_FIELDS 2
+static const char* const infoTypes[] = {"Version", "Stellar App"};
+static const char* const infoContents[] = {APPVERSION, "(c) 2022 Ledger"};
+
+#define NB_SETTINGS_SWITCHES 2
+#define SWITCH_IDX(token)    (token - SWITCH_HASH_SET_TOKEN)
+static uint8_t settings[NB_SETTINGS_SWITCHES] = {S_HASH_SIGNING_ENABLED, S_SEQUENCE_NUMBER_ENABLED};
+static nbgl_layoutSwitch_t switches[NB_SETTINGS_SWITCHES];
+
+void onQuitCallback(void) {
+    os_sched_exit(-1);
+}
+
+static bool settingsNavCallback(uint8_t page, nbgl_pageContent_t* content) {
+    if (page == 0) {
+        switches[0].text = "Hash signing";
+        switches[0].subText = "Enable transaction hash signing";
+        switches[0].token = SWITCH_HASH_SET_TOKEN;
+        switches[0].tuneId = TUNE_TAP_CASUAL;
+        switches[1].text = "Sequence number";
+        switches[1].subText = "Display sequence in transactions";
+        switches[1].token = SWITCH_SEQUENCE_SET_TOKEN;
+        switches[1].tuneId = TUNE_TAP_CASUAL;
+
+        content->type = SWITCHES_LIST;
+        content->switchesList.nbSwitches = NB_SETTINGS_SWITCHES;
+        content->switchesList.switches = (nbgl_layoutSwitch_t*) switches;
+    } else if (page == 1) {
+        content->type = INFOS_LIST;
+        content->infosList.nbInfos = NB_INFO_FIELDS;
+        content->infosList.infoTypes = (const char**) infoTypes;
+        content->infosList.infoContents = (const char**) infoContents;
+    } else {
+        return false;
+    }
+    return true;
+}
+
+static void settingsControlsCallback(int token, uint8_t index) {
+    UNUSED(index);
+    switch (token) {
+        case SWITCH_HASH_SET_TOKEN:
+        case SWITCH_SEQUENCE_SET_TOKEN:
+            switches[SWITCH_IDX(token)].initState = !(switches[SWITCH_IDX(token)].initState);
+            SETTING_TOGGLE(settings[SWITCH_IDX(token)]);
+            displaySettingsMenu();
+            break;
+        default:
+            PRINTF("Should not happen !");
+            break;
+    }
+}
+
+static void displaySettingsMenu(void) {
+    nbgl_useCaseSettings("Stellar settings",
+                         0,
+                         2,
+                         true,
+                         ui_menu_main,
+                         settingsNavCallback,
+                         settingsControlsCallback);
+}
+
+void ui_menu_main(void) {
+    switches[SWITCH_IDX(SWITCH_HASH_SET_TOKEN)].initState =
+        (HAS_SETTING(S_HASH_SIGNING_ENABLED)) ? ON_STATE : OFF_STATE;
+    switches[SWITCH_IDX(SWITCH_SEQUENCE_SET_TOKEN)].initState =
+        (HAS_SETTING(S_SEQUENCE_NUMBER_ENABLED)) ? ON_STATE : OFF_STATE;
```

Comment:
Can be moved to `settingsNavCallback` and then also removed from `settingsControlsCallback`

Reply:
Done.